packages/assets-controllers/src/multi-chain-accounts-service/multi-chain-accounts.ts

```diff
@@ -25,33 +26,56 @@ const getBalancesUrl = (
 
 /**
  * Fetches Supported Networks.
+ *
+ * @param options - options to pass down for a more refined search
+ * @param options.getAuthenticationControllerBearerToken - function to get the bearer token from the AuthenticationController
  * @returns supported networks (decimal)
  */
-export async function fetchSupportedNetworks(): Promise<number[]> {
+export async function fetchSupportedNetworks(options: {
+  getAuthenticationControllerBearerToken: () => ReturnType<
+    AuthenticationControllerGetBearerToken['handler']
+  >;
+}): Promise<number[]> {
   const url = new URL(`${MULTICHAIN_ACCOUNTS_DOMAIN}/v1/supportedNetworks`);
-  const response: GetSupportedNetworksResponse = await handleFetch(url);
+  const authenticationControllerBearerToken =
+    await options.getAuthenticationControllerBearerToken();
+  const response: GetSupportedNetworksResponse = await handleFetch(url, {
+    headers: {
+      Authorization: `Bearer ${authenticationControllerBearerToken}`,
+    },
+  });
   return response.fullSupport;
 }
 
 /**
  * Fetches Balances for multiple networks.
+ *
  * @param address - address to fetch balances from
  * @param options - params to pass down for a more refined search
  * @param options.networks - the networks (in decimal) that you want to filter by
+ * @param options.getAuthenticationControllerBearerToken - function to get the bearer token from the AuthenticationController
  * @param platform - indicates whether the platform is extension or mobile
  * @returns a Balances Response
  */
 export async function fetchMultiChainBalances(
   address: string,
-  options: { networks?: number[] },
+  options: {
+    getAuthenticationControllerBearerToken: () => ReturnType<
+      AuthenticationControllerGetBearerToken['handler']
+    >;
+    networks?: number[];
+  },
   platform: 'extension' | 'mobile',
 ) {
   const url = getBalancesUrl(address, {
     networks: options?.networks?.join(),
   });
+  const authenticationControllerBearerToken =
+    await options.getAuthenticationControllerBearerToken();
   const response: GetBalancesResponse = await handleFetch(url, {
     headers: {
       'x-metamask-clientproduct': `metamask-${platform}`,
+      Authorization: `Bearer ${authenticationControllerBearerToken}`,
```

Comment:
Very interesting!

Open discussion - could we pass the authorization header as a request/response cookie? All `.metamask.io` APIs will receive this cookie and can choose to use it?

It also prevents us from updating all the areas in the codebase?

Reply:
Yep great discussion, we can consider something once we support more authenticated APIs.